### PR TITLE
[WFLY-17658] adds support for wildfly channels dependency versions ov…

### PR DESCRIPTION
**Pull request by emmartins**
…erride
Issue: https://issues.redhat.com/browse/WFLY-17658

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -44,66 +44,49 @@
...
+    <dependencies>
+      <dependency>
+        <groupId>org.apache.maven.plugin-tools</groupId>
+        <artifactId>maven-plugin-tools-annotations</artifactId>
+        <version>${version.mavenPluginTools}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.wildfly.channel</groupId>
+        <artifactId>maven-resolver</artifactId>
+        <version>1.0.0.Final</version>
```
Comment (spyrkob) [inline pom.xml:63]: should this be `${version.wildFlyChannels}` ?

File: src/main/java/org/wildfly/plugins/bombuilder/BuildBomMojo.java [hunk 2]
```diff
@@ -225,6 +233,10 @@ public enum InheritExclusions {
     private final PomDependencyVersionsTransformer versionsTransformer;
     private final ModelWriter modelWriter;
 
+
+    @Parameter(alias = "channels", required = false)
+    private List<ChannelConfiguration> channels;
```
Comment (spyrkob) [inline src/main/java/org/wildfly/plugins/bombuilder/BuildBomMojo.java:238]: @emmartins `ChannelConfiguration` class is missing. I don't think it's one from channel library...

**Comment by emmartins** [comment]
@spyrkob fixed, somehow I failed to add it to the commit

**Comment by emmartins** [comment]
Sure, maybe make a PR on top of my branch with any fixes you need to make
it fully work?

File: pom.xml (hunk 1 above)
Comment (emmartins) [reply pom.xml]: yep, thanks for fixing that too

**Pull request merged**